### PR TITLE
fix(dev-server-rollup): support latest node package exports

### PR DESCRIPTION
**Pull request by daKmoR**
## What I did

1. Update dependency `@rollup/plugin-node-resolve` 13.x
2. Update DEV dependencies `rollup` & `@rollup/plugin-commonjs`

### CODE REVIEW

File: packages/dev-server-rollup/test/node/plugins/commonjs.test.ts [hunk 1]
```diff
@@ -116,20 +116,19 @@ exports.default = _default;`;
...
+        '/__web-dev-server__/rollup/commonjsHelpers.js?web-dev-server-rollup-null-byte=%00commonjsHelpers.js',
       );
-      expectIncludes(
-        text,
-        'var foo = commonjsHelpers.createCommonjsModule(function (module, exports) {',
-      );
-      expectIncludes(text, 'exports.__esModule = true;');
-      expectIncludes(text, 'exports.default = void 0;');
+      // expectIncludes(
+      //   text,
```
Comment (LarsDenBakker) [inline packages/dev-server-rollup/test/node/plugins/commonjs.test.ts:122]: Is this intentional? Then lets remove it.

File: packages/dev-server-rollup/test/node/plugins/commonjs.test.ts [hunk 2]
```diff
@@ -116,20 +116,19 @@ exports.default = _default;`;
...
       expectIncludes(text, "var _default = 'foo';");
-      expectIncludes(text, 'exports.default = _default;');
-      expectIncludes(
-        text,
-        'export default /*@__PURE__*/commonjsHelpers.getDefaultExportFromCjs(foo);',
-      );
+      // expectIncludes(
+      //   text,
+      //   'export default /*@__PURE__*/commonjsHelpers.getDefaultExportFromCjs(foo);',
+      // );
```
Comment (LarsDenBakker) [inline packages/dev-server-rollup/test/node/plugins/commonjs.test.ts:131]: Remove if intended

File: packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts [hunk 3]
```diff
@@ -129,7 +129,8 @@ describe('@rollup/plugin-node-resolve', () => {
     }
   });
 
-  it('node modules resolved outside root directory are rewritten with commonjs', async () => {
+  // works with @rollup/plugin-commonjs 22.x but not with 21.x which we use to test?
```
Comment (LarsDenBakker) [inline packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts:132]: Can we update to latest commonjs plugin?

File: packages/dev-server-rollup/test/node/plugins/commonjs.test.ts [hunk 4]
```diff
@@ -158,13 +157,13 @@ exports.default = _default;`;
...
-      expectIncludes(text, "import './bar.js';");
-      expectIncludes(
-        text,
-        "import bar from '/__web-dev-server__/rollup/bar.js?web-dev-server-rollup-null-byte=%00",
-      );
+      expectIncludes(text, 'import "./bar.js";');
+      // expectIncludes(
+      //   text,
+      //   "import bar from '/__web-dev-server__/rollup/bar.js?web-dev-server-rollup-null-byte=%00",
+      // );
```
Comment (LarsDenBakker) [inline packages/dev-server-rollup/test/node/plugins/commonjs.test.ts:164]: Remove if intended

File: packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts (hunk 3 above)
Comment (daKmoR) [reply packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts:132]: if we do then some other tests fail 🙈

Comment (daKmoR) [reply packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts:132]: e.g. then bare import to the commonjs helpers does not get rewritten... 😅

but v22 came out 10 hours ago... so maybe for next time

File: packages/dev-server-rollup/test/node/plugins/commonjs.test.ts (hunk 1 above)
Comment (daKmoR) [reply packages/dev-server-rollup/test/node/plugins/commonjs.test.ts:122]: jup intentional... left in so you can see that we will no longer test this implementation detail... not sure if it's an important implementation details...

if not then we can remove it

File: packages/dev-server-rollup/test/node/plugins/commonjs.test.ts (hunk 2 above)
Comment (daKmoR) [reply packages/dev-server-rollup/test/node/plugins/commonjs.test.ts]: removed

File: packages/dev-server-rollup/test/node/plugins/commonjs.test.ts (hunk 1 above)
Comment (LarsDenBakker) [reply packages/dev-server-rollup/test/node/plugins/commonjs.test.ts]: Yeah wr test impl details, but then we do know its doing its job. So we can remove this then

File: packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts (hunk 3 above)
Comment (LarsDenBakker) [reply packages/dev-server-rollup/test/node/plugins/node-resolve.test.ts:132]: Ok lets leave it then

**Comment by LarsDenBakker** [comment]
LGTM

**Comment by daKmoR** [comment]
superseded by https://github.com/modernweb-dev/web/pull/1942

**Pull request closed without merging**